contracts/utils/generalizedConvexStratDeployment.js

```diff
@@ -0,0 +1,179 @@
+const { deploymentWithProposal } = require("../utils/deploy");
+const { BigNumber } = require("ethers");
+
+module.exports = ({
+  deployName,
+  forceDeploy,
+  mainTokenName,
+  mainTokenSymbol,
+  rewardTokenNames,
+  assets,
+  pTokens,
+  platformAddress,
+  cvxDepositorAddress,
+  metapoolAddress,
+  metapoolLPToken,
+  mainTokenAddress,
+  cvxRewardStakerAddress,
+  cvxDepositorPTokenId,
+  redeployVault,
+}) => {
+  return deploymentWithProposal(
+    { deployName, forceDeploy },
+    async ({
+      assetAddresses,
+      deployWithConfirmation,
+      ethers,
+      getTxOpts,
+      withConfirmation,
+    }) => {
+      const { deployerAddr, governorAddr } = await getNamedAccounts();
+      const sDeployer = await ethers.provider.getSigner(deployerAddr);
+
+      // Current contracts
+      const cVaultProxy = await ethers.getContract("VaultProxy");
+      const cVaultAdmin = await ethers.getContractAt(
+        "VaultAdmin",
+        cVaultProxy.address
+      );
+      const cVault = await ethers.getContractAt("Vault", cVaultProxy.address);
+      const proxyName = `Convex${mainTokenSymbol}MetaStrategyProxy`;
+
+      // Deployer Actions
+      // ----------------
+
+      // 1. Deploy new proxy
+      // New strategy will be living at a clean address
+      const dConvexTokenMetaStrategyProxy = await deployWithConfirmation(
+        proxyName
+      );
+      const cConvexTokenMetaStrategyProxy = await ethers.getContractAt(
+        proxyName,
+        dConvexTokenMetaStrategyProxy.address
+      );
+
+      // 2. Deploy new implementation
+      const dConvexTokenMetaStrategyImpl = await deployWithConfirmation(
```

Comment:
I don't know that we need to deploy a new implementation for each strategy? Can we just reuse the same one?

Reply:
good point fixed here: https://github.com/OriginProtocol/origin-dollar/pull/1002/commits/ab1625fd5fb43c113642802f6c3e799ce4b49c3b